src/dbms/database.hpp

```diff
@@ -145,9 +147,11 @@ class Database {
   /**
    * @brief Returns the PlanCache vector raw pointer
    *
-   * @return utils::SkipList<query::PlanCacheEntry>*
+   * @return uutils::Synchronized<utils::LRUCache<uint64_t, std::shared_ptr<CachedPlan>>, utils::SpinLock>
    */
-  utils::SkipList<query::PlanCacheEntry> *plan_cache() { return &plan_cache_; }
+  utils::Synchronized<utils::LRUCache<uint64_t, std::shared_ptr<query::CachedPlan>>, utils::SpinLock> *plan_cache() {
+    return &plan_cache_;
```

Comment:
You can use here I think utils::RWSpinLock, I think it should be even better than utils::SpinLock